src/openlcb/TractionTrain.cxx

```diff
@@ -213,7 +213,20 @@ struct TrainService::Impl
                     uint16_t value = payload()[4];
                     value <<= 8;
                     value |= payload()[5];
-                    train_node()->train()->set_fn(address, value);
+                    bn_.reset(this);
+                    bool should_apply =
+                        train_node()->function_policy(nmsg()->src, payload()[0],
+                            address, value, bn_.new_child());
+                    if (!bn_.abort_if_almost_done())
```

Comment:
What is this "abort_if_almost_done()" about. Could you ad some more detailed commentary?

Reply:
done